datadog/resource_datadog_synthetics_test_.go

```diff
@@ -494,10 +507,130 @@ func syntheticsTestStep() *schema.Schema {
 					Type:        schema.TypeInt,
 					Optional:    true,
 				},
-				"params": {
-					Description: "Parameters for the step as JSON string.",
+				"params": &paramsSchema,
+			},
+		},
+	}
+}
+
+func syntheticsBrowserStepParams() schema.Schema {
+	return schema.Schema{
+		Description: "Parameters for the step.",
+		Type:        schema.TypeList,
+		MaxItems:    1,
+		Required:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"attribute": {
+					Description: `Name of the attribute to use for an "assert attribute" step.`,
 					Type:        schema.TypeString,
-					Required:    true,
+					Optional:    true,
+				},
+				"check": {
+					Description:  "Check type to use for an assertion step.",
+					Type:         schema.TypeString,
+					Optional:     true,
+					ValidateFunc: validateEnumValue(datadogV1.NewSyntheticsCheckTypeFromValue),
+				},
+				"click_type": {
+					Description:  `Type of click to use for a "click" step.`,
+					Type:         schema.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringInSlice([]string{"contextual", "double", "primary"}, false),
+				},
+				"code": {
+					Description: "Javascript code to use for the step.",
+					Type:        schema.TypeString,
+					Optional:    true,
+				},
+				"delay": {
+					Description: `Delay between each key stroke for a "type test" step.`,
+					Type:        schema.TypeInt,
+					Optional:    true,
+				},
+				"element": {
+					Description: "Element to use for the step, json encoded string.",
+					Type:        schema.TypeString,
+					Optional:    true,
+				},
+				"email": {
+					Description: `Details of the email for an "assert email" step.`,
+					Type:        schema.TypeString,
+					Optional:    true,
+				},
+				"file": {
+					Description: `For an "assert download" step.`,
+					Type:        schema.TypeString,
+					Optional:    true,
+				},
+				"files": {
+					Description: `Details of the files for an "upload files" step, json encoded string.`,
+					Type:        schema.TypeString,
+					Optional:    true,
+				},
+				"modifiers": {
+					Description: `Modifier to use for a "press key" step.`,
+					Type:        schema.TypeList,
+					Optional:    true,
+					Elem: &schema.Schema{
+						Type:         schema.TypeString,
+						ValidateFunc: validation.StringInSlice([]string{"Alt", "Control", "meta", "Shift"}, false),
+					},
+				},
+				"playing_tab_id": {
+					Description: "ID of the tab to play the subtest.",
+					Type:        schema.TypeString,
+					Optional:    true,
+				},
+				"request": {
+					Description: "Request for an API step.",
+					Type:        schema.TypeString,
+					Optional:    true,
+				},
+				"subtest_public_id": {
```

Comment:
For confirmation, is it expected we don't have a `subtest` field here?

Reply:
Same as above, the parameters are exact matches (except for casing) of what is needed in the backend.